controller/src/beerocks/master/tasks/dynamic_channel_selection_task.cpp

```diff
@@ -118,6 +120,12 @@ void dynamic_channel_selection_task::work()
         break;
     }
     case eState::TRIGGER_SCAN: {
+        if (m_scanning != NO_SCAN_IN_PROGRESS) {
```

Comment:
When moving to the `TRIGGER_SCAN` state from the `IDLE` state, we also initialize the `m_last_scan_try_timestamp` member. If now the task will block until the previous scan is complete, what impact does it have on this timeout?
Maybe it's better to remain in `IDLE` state? or even in `INIT`?

Reply:
good point, but I think it is better to move the `m_last_scan_try_timestamp` set point to where we actually trigger the scan - meaning the `TRIGGER_SCAN` state.
will move the set of `m_last_scan_try_timestamp` to `TRIGGER_SCAN` state - right after setting the `m_scanning_task_id`

Reply:
please re-review. provided a solution that is based on your suggestion.